### DISCUSSION TITLE
pgadmin style function execution in sql editor doesn't work

### DISCUSSION THREAD

**Discussion by GoldenPenguin** [dbeaver/dbeaver#209] [Q&A] [answered]
Hello,

The following query works in pgadmin, but not in DBeaver :
`do
$$
begin
        for i in 1..10000 loop
                perform someFunction(1);
        end loop;
end
$$`

The error given is the following :

> DBCException: SQL Error [42601]: ERREUR: chaîne entre guillemets dollars non terminée sur ou >près de « $$
> begin
>       for i in 1..10000 loop
>              perform someFunction(1);
>     end loop »
> Position: 4

I just found out the project and I didn't check the source code, but my blind guess would be that the query interpreter doesn't handle the syntax (maybe DBeaver has an other way of specifying the execution of an anonymous function?).

**Comment by serge-rider** [comment, marked as answer]
You can execute this query by selecting it and executing as single statement (ctrl+enter).
DBeaver doesn't yet support pgSQL syntax and can't detect block delimiters.
